sparse/src/KokkosSparse_Utils_mkl.hpp

```diff
@@ -62,9 +62,15 @@ inline void mkl_internal_safe_call(sparse_status_t mkl_status, const char *name,
   }
 }
 
+}  // namespace Impl
+}  // namespace KokkosSparse
+
```

Comment:
Macros aren't really associated with a scope right? I don't think this can make any difference with how we or users could use the macro.

Reply:
That's correct however if I define it outside of all namespace then I am sure that the function call inside the macro is correctly qualified. i.e. this is to make sure that a user can call it from anywhere in their code... I made silly mistakes before like forgetting `KokkosSparse::Impl::` in front of `mkl_internal_safe_call`

Reply:
I mean, whether the function gets resolved correctly or not only depends on the context where the macro is expanded, not where it's defined. To be really safe, in case the user does something very silly like redefine ``namespace KokkosSparse`` in their own code, this could call
```
::KokkosSparse::Impl::mkl_internal_safe_call(...)
```
instead, then it should be safe to expand anywhere

Reply:
Yes, that's correct, the scope in which the macro is defined does not change how it gets expended so I guess I really don't need to take it out of the namespace.
Regarding the users redefining our namespace in their code, at that point they should be willing to suffer the consequences! : )